CDTDatastore/touchdb/TDReplicator.m

```diff
@@ -819,9 +907,14 @@ - (void)saveLastSequence
                       // read-only and don't attempt to read its checkpoint next time.
                   } else if (_db) {
                       id rev = response[@"rev"];
+                      id ID = response[@"id"];
                       if (rev) body[@"_rev"] = rev;
+                      if (ID) body[@"_id"] = ID;
                       self.remoteCheckpoint = body;
-                      [_db setLastSequence:_lastSequence withCheckpointID:checkpointID];
+                      if (ID && [self.db saveCheckpointDocument:body error:nil]) {
+                          CDTLogWarn(CDTREPLICATION_LOG_CONTEXT,
```

Comment:
I'm not sure, it is a though one, I think we perhaps should error out, but with a specific error to indicate that the checkpoint wasn't written. I'm not sure if this is something we could do without a major release though.